### ISSUE TITLE
How do you normalize pixel values of each band?

### ISSUE THREAD

**Issue by xueleichen** [phelber/EuroSAT#1]
Hi,

I notice that images in EuroSAT (RGB) are of uint8 datatype with the range [0,256]. 

However images in EuroSAT (MS) are of uint16 datatype. And the range is different across different bands and different images. 

E.g. 
**AnnualCrop_1.tif** band02 [1002,1534], band03 [905,1664]
**Forest_1.tif** band02 [776,1009]

How do you normalize these multispectral images to the range like uint8 [0,256] or float [0,1]?

Thank you!

**Comment by xueleichen** [comment]
The multispectral images of Sentinel-2 are 12-bit. That means the range of pixel value is [0,4095]. Then we can do the linear normalization

Source: https://sentinel.esa.int/web/sentinel/missions/sentinel-2/instrument-payload/resolution-and-swath

**Comment by johannespetrat** [comment]
Hi @xueleichen, I'm currently facing the same issue and tried rescaling the images to the range [0, 255] by multiplying by a factor of 255/4095. Is that what you meant with "linear normalization"?
The images are still slightly different (see screenshot below). Could you please elaborate on your solution to this?
![example](https://user-images.githubusercontent.com/18034999/134013913-68d98b18-9a12-49d7-b44d-d391919c79a0.png)

**Comment by xueleichen** [comment]
Hi @johannespetrat , I checked my previous experiment python code. These are what I did:
```
img = rasterio.open(path)
a = img.read([4,3,2,5,6,7,8,9,12,13])/4095.
a=a.transpose((1,2,0)) # CHW -> HWC
X[i,:,:,:] = np.clip(a, 0, 1)
```
Yes, I think I mean multiplying a constant 1/4095  or 255/4095 by 'linear normalization'. To be honest, I did not compare RGB generated from multispectral and RGB provided by the author.

One more thing I want to mention is that, the order of RGB bands in multispectral data is not by [R,G,B], as you can see from my previous code. Actually, Band 2 is Blue, Band 3 is Green, and Band 4 is Red.

How do you visualize the above image using bands from multispectral data? Could you please check that in your code?

**Comment by johannespetrat** [comment]
Thanks for getting back to me so quickly @xueleichen ! I used the same logic as in your code snippet (incl. swapping the bands 2 and 4) to generate the RGB images, and it seems that the author used a different method because I can't reproduce their images. 
In the end, I retrained my model with the multispectral data (instead of the RGB images), so it's reassuring that you used the same approach.
